### ISSUE TITLE
Find in Folder doesn't work correctly when the path contains certain characters

### ISSUE THREAD

**Issue by anthonyvdotbe** [microsoft/vscode#174538]
Type: <b>Bug</b>

To reproduce:

1. add 3 folders named `one`, `(two)`, `[three]`
2. in each folder add a file containing `foo`
3. do "Find in Folder" on each of the folders & search for `foo`

Expected: there's always exactly 1 result
Actual: for folders `(two)` and `[three]`, there are 3 results.

Note: there's probably more problematic characters (possibly all that have a special meaning in regular expressions?) and it doesn't matter if they are in the leaf directory or any parent directory

VS Code version: Code - Insiders 1.76.0-insider (dc619de8d9efdabd781fc5658d0051813c347055, 2023-02-16T05:23:44.671Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10610U CPU @ 1.80GHz (8 x 2304)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.61GB (4.12GB free)|
|Process Argv|--crash-reporter-id 072fc852-4655-4a25-b52d-8b51a987e2f7|
|Screen Reader|no|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30422396
pythontb:30258533
pythonptprofiler:30281269
vsdfh931cf:30280410
vshan820:30294714
pythondataviewer:30285072
vscod805:30301674
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30404738
cppdebug:30492333
vsclangdf:30492506
c4g48928:30535728
dsvsc012:30540252
pynewext54:30618038
pylantcb52:30590116
pyindex848:30611229
nodejswelcome1:30587009
pyind779:30611226
pythonsymbol12:30651887
showlangstatbar:30659908
pythonb192cf:30661257

```

</details>

**Comment by andreamah** [comment]
I'm a bit confused about why there should be 3 results in `(two)` and `[three]` if step 2 of your repro steps stated to add one file containing foo to each folder. Then, if I asked to find in folder for each individual folder, wouldn't there be one result in each?

**Comment by anthonyvdotbe** [comment]
To rephrase more clearly, first create the following file structure:

* root
  * one
    * foo.txt
  * (two)
    * foo.txt
  * [three]
    * foo.txt

where all `foo.txt` files contain the string `foo`.

Then, for each of `one`, `(two)`, and `[three]`, do "Find in Folder" on it & search for `foo`.

> I'm a bit confused about why there should be 3 results [...] wouldn't there be one result in each?

Indeed: the expected behavior is that there's always exactly 1 result, but the actual behavior is that, for `(two)` and `[three]`, there are 3 results. The problem is that the `files to include` field, which is set to the directory path when choosing `Find in Folder`, is ignored during the search in some cases. And so the search incorrectly finds all 3 `foo.txt` files, instead of just the one in the chosen folder.

**Comment by anthonyvdotbe** [comment]
@andreamah I assume the `info-needed` label can be removed? Or is there any other info I should provide?

**Comment by andreamah** [comment]
I can't seem to reproduce this issue? Am I supposed to be seeing 3 results somewhere?

https://user-images.githubusercontent.com/31675041/224415093-43908e11-54b7-4f81-847e-f323552d8958.mp4

**Comment by anthonyvdotbe** [comment]
Duplicate of #82415, fixed in 1.76.1
